server/controllers/payroll/multiplePayrollIndice/paramter.config.js

```diff
@@ -21,19 +22,251 @@ function detail(req, res, next) {
 }
 
 // settup staffing indice parameters
-function create(req, res, next) {
+async function create(req, res, next) {
   const data = req.body;
+
   data.uuid = db.uuid();
   const id = req.body.payroll_configuration_id;
+
   const transaction = db.transaction();
+  const minMonentaryUnit = req.session.enterprise.min_monentary_unit;
+  const payEnvelope = req.body.pay_envelope;
+  const workingDays = req.body.working_days;
+  const requestsUpdateIndices = await stagePaymentIndice(id);
+
+  const paymentIndice = requestsUpdateIndices[0];
+  const employeesGradeIndice = requestsUpdateIndices[1];
+  const aggregateOtherProfits = requestsUpdateIndices[2];
+  const rubricMonetaryValue = requestsUpdateIndices[3];
+
+  let totalCode = 0;
+  let rubricPayRateId;
+  let rubricGrossSallaryId;
+  let rubricNumberOfDaysId;
 
   transaction.addQuery('DELETE FROM staffing_indice_parameters WHERE payroll_configuration_id =?', id);
   transaction.addQuery('INSERT INTO staffing_indice_parameters SET ?', data);
-  transaction.addQuery('CALL updateIndices(?)', id);
+
+  const updateStaffingIndice = `
+    UPDATE stage_payment_indice SET rubric_value = ?
+    WHERE payroll_configuration_id = ? AND employee_uuid = ? AND rubric_id = ?;
+  `;
+
+  const updateEmployeeIndividualySalary = `
+    UPDATE employee SET individual_salary = ? WHERE uuid = ?
+  `;
+
+  employeesGradeIndice.forEach(emp => {
+    let totalDays = 0;
+    let rubricTotalDaysId;
+    let rubricReagisteredIndexId;
+    let rubricTotalCodeId;
+    let rubricDayIndexId;
+
+    emp.totalBase = 0;
+
+    transaction.addQuery(`DELETE FROM employee_advantage WHERE employee_uuid = ?`, [emp.employee_buid]);
+
+    paymentIndice.forEach(ind => {
```

Comment:
This would be more efficient if you use .find()

Reply:
I can't quite figure out which functionality I should replace with the find() function

Reply:
It looks like you have one item and you're looping through an array  (for foreEach) until you find a match.  Then you do the computations on the matching item.  And then you loop through the rest of the items in the array.   I was suggesting, you search for the matching item with find() and then do the operation on it.   find() should be faster than looping through the entire array.   Maybe I misunderstood what you're doing here...